Core/GDCore/String.cpp

```diff
@@ -251,26 +251,28 @@ String& String::replace_if(iterator i1, iterator i2, std::function<bool(char32_t
     return *this;
 }
 
-String& String::RemoveConsecutiveOccurrences(iterator i1, iterator i2, const char c)
-{
-    std::vector<std::pair<size_type, size_type>> ranges_to_remove;
-    for(iterator current_index = i1.base(); current_index < i2.base(); current_index++)
-    {
-        if (*current_index == c){
-            iterator current_subindex = current_index;
-            std::advance(current_subindex, 1);
-            if (*current_subindex == c) {
-                while(current_subindex < end() && *current_subindex == c)
-                {
-                    current_subindex++;
-                }
-                replace(std::distance(begin(), current_index),
-                        std::distance(current_index, current_subindex),
-                        c);
-
-                std::advance(current_index, 1);
-            }
+String &String::RemoveConsecutiveOccurrences(iterator i1,
+                                             iterator i2,
+                                             const char c) {
+    iterator end = i2;
+    for (iterator current_index = i1.base(); current_index < end.base();
+         current_index++) {
+      if (*current_index == c) {
+        iterator current_subindex = current_index;
+        current_subindex++;
+        if (*current_subindex == c) {
+          while (current_subindex < end.base() && *current_subindex == c) {
+            current_subindex++;
+          }
+        }
```

Comment:
In other words, is it possible to do something like:
```suggestion
          while (current_subindex < end.base() && *current_subindex == c) {
            current_subindex++;
          }
```